`IncompletePostTestCase` is failing a lot on Windows

### Description

Just to mention a few recent failures:
- https://github.com/quarkusio/quarkus/pull/23626#issuecomment-1036950289
- https://github.com/quarkusio/quarkus/pull/23587#issuecomment-1035423206
- https://github.com/quarkusio/quarkus/pull/23583#issuecomment-1035257234
- https://github.com/quarkusio/quarkus/pull/23576#issuecomment-1034911742

```
[ERROR] io.quarkus.resteasy.test.IncompletePostTestCase.testIncompleteWrite  Time elapsed: 0.845 s  <<< ERROR!
java.net.ConnectException: Connection refused: connect
	at java.base/java.net.PlainSocketImpl.connect0(Native Method)
	at java.base/java.net.PlainSocketImpl.socketConnect(PlainSocketImpl.java:101)
	at java.base/java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:412)
	at java.base/java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:255)
	at java.base/java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:237)
	at java.base/java.net.SocksSocketImpl.connect(SocksSocketImpl.java:392)
	at java.base/java.net.Socket.connect(Socket.java:609)
	at java.base/java.net.Socket.connect(Socket.java:558)
	at java.base/java.net.Socket.<init>(Socket.java:454)
	at java.base/java.net.Socket.<init>(Socket.java:231)
	at io.quarkus.resteasy.test.IncompletePostTestCase.testIncompleteWrite(IncompletePostTestCase.java:36)
```
It starts with:
```
2022-02-11 22:54:16,687 ERROR [io.qua.ver.htt.run.QuarkusErrorHandler] (executor-thread-117) HTTP Request to /post failed, error id: fcbec178-715a-4b0b-8981-62a6ee53fdf1-1: javax.enterprise.context.ContextNotActiveException
	at io.quarkus.arc.impl.ClientProxies.getDelegate(ClientProxies.java:46)
	at io.quarkus.security.runtime.SecurityIdentityAssociation_ClientProxy.arc$delegate(Unknown Source)
	at io.quarkus.security.runtime.SecurityIdentityAssociation_ClientProxy.setIdentity(Unknown Source)
	at io.quarkus.resteasy.runtime.standalone.VertxRequestHandler.dispatch(VertxRequestHandler.java:124)
	at io.quarkus.resteasy.runtime.standalone.VertxRequestHandler$1.run(VertxRequestHandler.java:91)
	at io.quarkus.vertx.core.runtime.VertxCoreRecorder$13.runWith(VertxCoreRecorder.java:543)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2449)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1462)
	at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:29)
	at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:29)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
```
Then things like:
```
2022-02-11 22:54:16,695 ERROR [io.qua.ver.htt.run.QuarkusErrorHandler] (executor-thread-194) HTTP Request to /post failed, error id: fcbec178-715a-4b0b-8981-62a6ee53fdf1-190: java.lang.NullPointerException
	at io.quarkus.resteasy.runtime.standalone.VertxRequestHandler.dispatch(VertxRequestHandler.java:117)
	at io.quarkus.resteasy.runtime.standalone.VertxRequestHandler$1.run(VertxRequestHandler.java:91)
	at io.quarkus.vertx.core.runtime.VertxCoreRecorder$13.runWith(VertxCoreRecorder.java:543)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2449)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1462)
	at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:29)
	at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:29)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:829)
```
and:
```
2022-02-11 22:54:16,814 ERROR [io.ver.ext.web.RoutingContext] (executor-thread-6) Unhandled exception in router: java.util.concurrent.RejectedExecutionException: event executor terminated
	at io.netty.util.concurrent.SingleThreadEventExecutor.reject(SingleThreadEventExecutor.java:923)
	at io.netty.util.concurrent.SingleThreadEventExecutor.offerTask(SingleThreadEventExecutor.java:350)
	at io.netty.util.concurrent.SingleThreadEventExecutor.addTask(SingleThreadEventExecutor.java:343)
	at io.netty.util.concurrent.SingleThreadEventExecutor.execute(SingleThreadEventExecutor.java:825)
	at io.netty.util.concurrent.SingleThreadEventExecutor.execute(SingleThreadEventExecutor.java:815)
	at io.vertx.core.net.impl.ConnectionBase.queueForWrite(ConnectionBase.java:238)
	at io.vertx.core.net.impl.ConnectionBase.writeToChannel(ConnectionBase.java:228)
	at io.vertx.core.net.impl.ConnectionBase.writeToChannel(ConnectionBase.java:218)
	at io.vertx.core.net.impl.ConnectionBase.writeToChannel(ConnectionBase.java:214)
	at io.vertx.core.http.impl.Http1xServerResponse.end(Http1xServerResponse.java:416)
	at io.vertx.core.http.impl.Http1xServerResponse.end(Http1xServerResponse.java:391)
	at io.vertx.core.http.impl.Http1xServerResponse.end(Http1xServerResponse.java:370)
	at io.quarkus.vertx.http.runtime.QuarkusErrorHandler.writeResponse(QuarkusErrorHandler.java:191)
	at io.quarkus.vertx.http.runtime.QuarkusErrorHandler.jsonResponse(QuarkusErrorHandler.java:177)
	at io.quarkus.vertx.http.runtime.QuarkusErrorHandler.handle(QuarkusErrorHandler.java:161)
	at io.quarkus.vertx.http.runtime.QuarkusErrorHandler.handle(QuarkusErrorHandler.java:30)
	at io.vertx.ext.web.impl.RouteState.handleFailure(RouteState.java:1218)
	at io.vertx.ext.web.impl.RoutingContextImplBase.iterateNext(RoutingContextImplBase.java:160)
	at io.vertx.ext.web.impl.RoutingContextImpl.next(RoutingContextImpl.java:141)
	at io.vertx.ext.web.impl.RoutingContextImpl.doFail(RoutingContextImpl.java:604)
	at io.vertx.ext.web.impl.RoutingContextImpl.fail(RoutingContextImpl.java:192)
	at io.vertx.ext.web.impl.RoutingContextImpl.fail(RoutingContextImpl.java:181)
	at io.quarkus.resteasy.runtime.standalone.VertxRequestHandler$1.run(VertxRequestHandler.java:93)
	at io.quarkus.vertx.core.runtime.VertxCoreRecorder$13.runWith(VertxCoreRecorder.java:543)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2449)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1462)
	at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:29)
	at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:29)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:829)
```
(+ maybe more, I stopped at this point)

### Implementation ideas

_No response_

--- a/extensions/resteasy-classic/resteasy/deployment/src/test/java/io/quarkus/resteasy/test/IncompletePostTestCase.java
+++ b/extensions/resteasy-classic/resteasy/deployment/src/test/java/io/quarkus/resteasy/test/IncompletePostTestCase.java
@@ -8,8 +8,6 @@ import org.apache.http.params.CoreConnectionPNames;
 import org.hamcrest.Matchers;
 import org.junit.jupiter.api.Assertions;
 import org.junit.jupiter.api.Test;
-import org.junit.jupiter.api.condition.DisabledOnOs;
-import org.junit.jupiter.api.condition.OS;
 import org.junit.jupiter.api.extension.RegisterExtension;
 
 import io.quarkus.test.QuarkusUnitTest;
@@ -18,13 +16,13 @@ import io.restassured.RestAssured;
 import io.restassured.config.HttpClientConfig;
 import io.restassured.config.RestAssuredConfig;
 
-@DisabledOnOs(value = OS.WINDOWS, disabledReason = "very flaky, see https://github.com/quarkusio/quarkus/issues/23642")
 public class IncompletePostTestCase {
 
     @RegisterExtension
     static QuarkusUnitTest runner = new QuarkusUnitTest()
             .withApplicationRoot((jar) -> jar
-                    .addClasses(PostEndpoint.class));
+                    .addClasses(PostEndpoint.class))
+            .overrideConfigKey("quarkus.http.accept-backlog", "200");
 
     @TestHTTPResource
     URL url;
@@ -34,8 +32,8 @@ public class IncompletePostTestCase {
         PostEndpoint.invoked = false;
 
         //make sure incomplete writes do not block threads
-        //and that incoplete data is not delivered to the endpoint
-        for (int i = 0; i < 1000; ++i) {
+        //and that incomplete data is not delivered to the endpoint
+        for (int i = 0; i < 100; ++i) {
             Socket socket = new Socket(url.getHost(), url.getPort());
             socket.getOutputStream().write(
                     "POST /post HTTP/1.1\r\nHost: localhost\r\nContent-length:10\r\n\r\ntest".getBytes(StandardCharsets.UTF_8));

--- a/extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/HttpConfiguration.java
+++ b/extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/HttpConfiguration.java
@@ -182,6 +182,12 @@ public class HttpConfiguration {
     public boolean tcpFastOpen;
 
     /**
+     * The accept backlog, this is how many connections can be waiting to be accepted before connections start being rejected
+     */
+    @ConfigItem
+    public int acceptBacklog;
+
+    /**
      * Path to a unix domain socket
      */
     @ConfigItem(defaultValue = "/var/run/io.quarkus.app.socket")

--- a/extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/VertxHttpRecorder.java
+++ b/extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/VertxHttpRecorder.java
@@ -844,6 +844,7 @@ public class VertxHttpRecorder {
         options.setReusePort(httpConfiguration.soReusePort);
         options.setTcpQuickAck(httpConfiguration.tcpQuickAck);
         options.setTcpCork(httpConfiguration.tcpCork);
+        options.setAcceptBacklog(httpConfiguration.acceptBacklog);
         options.setTcpFastOpen(httpConfiguration.tcpFastOpen);
         options.setCompressionSupported(httpConfiguration.enableCompression);
         options.setDecompressionSupported(httpConfiguration.enableDecompression);

